pkg/components/azure-arc-onboarding/component_test.go

```diff
@@ -0,0 +1,151 @@
+	t.Parallel()
+
+	c := azurearconboarding.NewConfig()
+
+	tests := []struct {
+		desc    string
+		config  string
+		wantErr bool
+	}{
+		{
+			desc: "Valid config",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+		},
+		{
+			desc: "Empty config",
+			config: `
+component azure-arc-onboarding {}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "All fields set to empty string",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = ""
+  tenant_id             = ""
+  application_password  = ""
+  resource_group        = ""
+  cluster_name          = ""
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty application_client_id",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = ""
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty tenant_id",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = ""
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty application_password",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "tenant-id"
+  application_password  = ""
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty resource_group",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = ""
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty cluster_name",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = ""
+}
+			`,
+			wantErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			body, diagnostics := util.GetComponentBody(test.config, azurearconboarding.Name)
+			if diagnostics.HasErrors() {
+				t.Errorf("Error getting component body: %v", diagnostics)
+			}
+
+			diagnostics = c.LoadConfig(body, &hcl.EvalContext{})
+			if test.wantErr != diagnostics.HasErrors() {
```

Comment:
I think comparing 2 booleans with `!=` is hard to read. With `&&`, it reads as `if we want error and diagnostics has no errors)`, as oppose to `if we want error is not equal to diagnostics has errors` :smile: 
```suggestion
			if test.wantErr && !diagnostics.HasErrors() {
```